arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,242 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
+3. [Checking the validity of your contract](./stylus-quickstart#checking-if-your-stylus-project-is-valid)
+4. [Deploying your contract](./stylus-quickstart#deploying-your-contract)
+5. [Exporting your contract's ABIs](./stylus-quickstart#exporting-solidity-abis)
+
+## Setting up your development environment
+
+### Prerequisites
 
 #### Rust toolchain
 
-Follow the instructions on [Rust Lang’s installation page](https://www.rust-lang.org/tools/install) to get a full Rust toolchain installed on your system. Make sure after installation that you have access to the programs `rustup`, `rustc`, and `cargo` from your preferred command line terminal (programs should be added to your system’s PATH, more instructions available on Rust’s website)
+Follow the instructions on [Rust Lang’s installation page](https://www.rust-lang.org/tools/install) to install a complete Rust toolchain on your system. After installation, ensure you have access to the programs `rustup`, `rustc`, and `cargo` from your preferred terminal application.
 
 #### VS Code
 
-We recommend VSCode as the IDE of choice for developing Stylus contracts for its excellent Rust support. See **[code.visualstudio.com](https://code.visualstudio.com/)** to install. Feel free to use another text editor or IDE if you’re comfortable with those.
+We recommend [VSCode](https://code.visualstudio.com/) as the IDE of choice for its excellent Rust support, but feel free to use another text editor or IDE if you’re comfortable with those.
 
 Some helpful VS Code extensions for Rust development:
 
-- [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer)
-- [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens)
-- [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml)
-- [Dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi)
+- [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer): Provides advanced features like smart code completion and on-the-fly error checks.
+- [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens): Immediately highlights errors and warnings in your code.
+- [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml): Improves syntax highlighting and other features for TOML files, often used in Rust projects.
+- [Dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi): Helps manage Rust crate versions directly from the editor.
   
-#### Testnet ETH for deployment
+#### Docker
 
-You’ll need some testnet ETH for deploying your Rust contract for live testing. Explained below in further detail.
+`cargo-stylus` uses Docker to run a local <a data-quicklook-from="arbitrum">Arbitrum</a> chain for testing. You can download Docker from [Docker’s website](https://www.docker.com/products/docker-desktop).
```

Comment:
This docker image doesn't run a local Arbitrum chain. It installs and runs cargo-stylus in a reproducible environment (cargo-stylus and rust versions specified by the user, but in a specific architecture)